### DISCUSSION TITLE
An unexpected error occurred invoking 'ChangeState' on the server

### DISCUSSION THREAD

**Discussion by nyannoying1337** [ppy/osu#15436] [Q&A]
Make a MP lobby > Start and press ESC > unexpected error occurred

[runtime.log](https://github.com/ppy/osu/files/7464070/runtime.log)

https://user-images.githubusercontent.com/17437302/139953904-82ea51db-0bc9-4b09-ad43-4ba3d4c2b733.mp4

**Comment by bdach** [comment]
This looks like the true cause:

```
2021-11-02 21:14:19 [important]: One or more errors occurred. (An unexpected error occurred invoking 'ChangeState' on the server. NotJoinedRoomException: This user has not yet joined a multiplayer room.)
```

Somehow the server claims that you had never joined the room in the first place, therefore the `RoomID` here is presumably null:

https://github.com/ppy/osu/blob/47914c49bacb3ea879e1045693757e7196916ac0/osu.Game/Screens/OnlinePlay/Lounge/LoungeSubScreen.cs#L252-L253

Can you reliably repeat that every time you create a room, or was this an isolated non-reproducible incident?

**Reply by nyannoying1337** [reply]
i can repeat that every time

**Comment by peppy** [comment]
Likely https://github.com/ppy/osu/pull/14203
